pkg/nubuild/build/build.go

```diff
@@ -82,11 +85,40 @@ type buildStep struct {
 	Func    func() error
 }
 
-func NewBuilder(parentLogger nuclio.Logger, options *Options) *Builder {
+func isURL(s string) bool {
+	return strings.HasPrefix(s, "http://") || strings.HasPrefix(s, "https://")
+}
+
+func findFunctionPath(options *Options, path string) error {
```

Comment:
This function should be a method of Builder. We should not mix OO semantics with procedural ones. 
Packages should *either* contain functions (i.e. common/url) or they should contain objects through which functionality is consumed. Since Builder is the latter, there should only be:
1. Public methods (with a pointer receiver)
2. Public "static" methods - used by other modules when you need to operate on the "object" without an instance
3. Private methods (with a pointer receiver)

Following this simple guideline will allow the code to be structured more consistently and no unholy half object half function packages will be born.